### PR TITLE
Persist PATH changes

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by isuruf** [comment]
@beckermr, any ideas why this fails on linux?

**Comment by beckermr** [comment]
We are mapping directories from outside the container. Maybe the permissions are not right? I wonder if we should copy the recipe dir to a temp dir and then try again. 

If that works, note that we will need to ship a fix to smithy.

**Comment by beckermr** [comment]
I think this is working. We'll have to make a PR on smithy.

**Comment by beckermr** [comment]
Well I fixed the first one, but now the conda-build one is being annoying. Let me try to fix that too.

**Comment by beckermr** [comment]
It a straight setuptools install works, then we should backout my changes.

**Comment by beckermr** [comment]
I don't understand why pip even sees the build env version. That feels like some weird bug.

**Comment by beckermr** [comment]
OK @isuruf just changing the docker mount to `rw` from `ro` fixed this without any of the other changes. Do you know why it was set at read only in the first place?

File: .scripts/build_steps.sh [hunk 1]
```diff
@@ -22,7 +22,9 @@ CONDARC
 conda install --yes --quiet conda-forge-ci-setup=2 conda-build pip -c conda-forge
 
 conda uninstall --quiet --yes --force conda-forge-ci-setup
-pip install --no-deps ${RECIPE_ROOT}/.
+pushd $(mktemp -d)
+python setup.py install --single-version-externally-managed --record record.txt
```
Comment (beckermr) [inline .scripts/build_steps.sh:26]: You would need to copy the source here.

**Comment by beckermr** [comment]
I made a PR on smithy here https://github.com/conda-forge/conda-smithy/pull/1303

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
OK to merge @isuruf? The change in the docker mount is in smithy so this will work out of the box once we get a smithy release.

**Pull request merged**